### DISCUSSION TITLE
F-keys / shortcuts

### DISCUSSION THREAD

**Discussion by nitram00** [fedorovvl/tso_client#170] [Ideas]
I like "new" settings/options but there's one thing I'm missing.
I don't use F8 (Timed production) very much so I would like to be able to re-map F8 or assign other shortcut (like CTRL+F4 for collies) to Generals so I can send/chose adv. templates faster.
Thank for all your work & in case there's already an option like this and I missed it I'm sorry.
Also if there's a way to add shortcut for e.g. rarity provision house (never understood how BB/UBI didn't add this one)...

**Comment by fedorovvl** [comment]
for custom bindings u can create simple userscript with smth like this
```
menu.addKeybBind(handler_func, key_code, ctrl_state, true);
```
key_code is uint button code.. check https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html
handler_func is function that run when u push buttons, for generals it _exudGeneralsMenuHandler
ctrl_state is boolean state of ctrl button (true or false)
last arg always true for user bindings
so to set Ctrl+F6 to generals u need to write this
```
menu.addKeybBind(_exudGeneralsMenuHandler, 117, true, true);
```
or just F10
```
menu.addKeybBind(_exudGeneralsMenuHandler, 121, false, true);
```
if combination binded already you'll see warning message
